### PR TITLE
[SW-786] Refactor: Use composition api to handle modals state

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

**Review by konradodo — APPROVED** [review]

File: src/popup/components/InputSelectAsset.vue [hunk 1]
```diff
@@ -25,31 +28,37 @@ export default {
...
   },
-  computed: {
-    displayToken() {
-      if (!this.value) return '';
-      const { symbol } = this.value;
+  setup(props, { emit }) {
+    const { openModal } = useModals();
+
+    const displayToken = computed(() => {
+      if (!props.value) return '';
```
Comment (pstachel) [inline src/popup/components/InputSelectAsset.vue:35]:
```suggestion
      if (!props.value) {
          return '';
      }
```

File: src/popup/components/InputSelectAsset.vue [hunk 2]
```diff
@@ -25,31 +28,37 @@ export default {
...
-    openAssetSelector() {
-      if (this.disabled) return;
+    });
+
+    function handleChange(token: IToken) {
+      emit('input', token);
+    }
+
+    function openAssetSelector() {
+      if (props.disabled) return;
```
Comment (pstachel) [inline src/popup/components/InputSelectAsset.vue:45]:
```suggestion
      if (props.disabled) {
          return;
      }
```

File: src/popup/components/InputSelectAsset.vue [hunk 3]
```diff
@@ -25,31 +28,37 @@ export default {
...
+
+      openModal(MODAL_ASSET_SELECTOR, {
+        selectedToken: props.value,
+        showTokensWithBalance: props.showTokensWithBalance,
+        resolve: (token) => token,
+      })
+        .then((token) => handleChange(token));
```
Comment (pstachel) [inline src/popup/components/InputSelectAsset.vue:52]: remove additional space
```suggestion
     .then((token) => handleChange(token));
```

File: src/popup/pages/InviteClaim.vue [hunk 4]
```diff
@@ -2,39 +2,41 @@
...
         });
-        return;
+      } catch (error: any) {
+        if (error.message === 'Invalid checksum') {
+          await openDefaultModal({
+            msg: 'The invite link is broken',
+          });
+          return;
+        }
+        if (await root.$store.dispatch('invites/handleNotEnoughFoundsError', { error, isInviteError: true })) return;
```
Comment (pstachel) [inline src/popup/pages/InviteClaim.vue:34]:
```suggestion
        if (await root.$store.dispatch('invites/handleNotEnoughFoundsError', { error, isInviteError: true })) {
        return;
        }
```

**Review by pstachel — APPROVED** [review]

File: src/popup/components/TransferSendForm.vue [hunk 5]
```diff
@@ -454,12 +451,13 @@ export default defineComponent({
...
+      openModal(MODAL_PAYLOAD_FORM, {
         payload: formModel.value.payload,
-      }).then((text) => {
-        formModel.value.payload = text;
-      }).catch(() => null);
+      })
+        .then((text) => {
+          formModel.value.payload = text;
+        })
+        .catch(() => null);
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/popup/components/TransferSendForm.vue:460]: Nope. Why you would like this change? Even the IDE is complaining after this change.

File: src/popup/components/InputSelectAsset.vue (hunk 3 above)
Comment (peronczyk) [reply src/popup/components/InputSelectAsset.vue:52]: Same comment as the one here: https://github.com/aeternity/superhero-wallet/pull/1948#discussion_r1147658838

**Pull request merged**